include/unifex/let_error.hpp

```diff
@@ -166,31 +171,47 @@ class _frcvr<Source, Func, Receiver, Error>::type {
 
   template(typename... Values)
     (requires receiver_of<Receiver, Values...>)
-  void set_value(Values&&... values) noexcept(
+  // Taking by value here to force a move/copy on the offchance the value
+  // objects live in the operation state, in which case destroying the
+  // predecessor operation state would invalidate it.
+  void set_value(Values... values) noexcept(
       is_nothrow_receiver_of_v<Receiver, Values...>) {
-    UNIFEX_ASSERT(op_ != nullptr);
-    unifex::set_value(std::move(op_->receiver_), (Values&&)values...);
+    // local copy, b/c cleanup deletes this
+    auto op = op_;
+    cleanup(op);
+    unifex::set_value(std::move(op->receiver_), std::move(values)...);
```

Comment:
Unrelated to the memory error you're fixing, but, given this implementation of `set_value` is `noexcept`, this should probably be wrapped in a `try`/`catch` that forwards `current_exception` to `set_error`.
```suggestion
    UNIFEX_TRY {
      unifex::set_value(std::move(op->receiver_), std::move(values)...);
    }
    UNIFEX_CATCH(...) {
      unifex::set_error(std::move(op->receiver_), std::current_exception());
    }
```